advanced/Scripts/query.sh

```diff
@@ -58,12 +61,131 @@ scanList(){
     esac
 }
 
+# function to count new domain entries since last gravity run
+CountNewDomains() {
+  printf %s "  ${INFO} Running query, please be patient..."
+  # store query result (count) in variable
+  newdomainentries=$(
+    sqlite3 "${gravityDBfile}" << EOSQL
+      ATTACH '${gravityOLDfile}' AS old;
+      SELECT count(DISTINCT domain) FROM gravity WHERE domain NOT IN (SELECT domain FROM old.gravity);
+EOSQL
+      )
+  printf '\r%s\n' "$(tput el)  ${INFO} ${newdomainentries} new domain(s) found."
+  exit 0
+}
+
+# function to list new domain entries since last gravity run
+ListNewDomains() {
+  if [ -z "${all}" ]; then
+    QueryLimit="Limit 100"
+  fi
```

Comment:
```suggestion
  QueryLimit=""
  if [ -z "${all}" ]; then
    QueryLimit="Limit 100"
  fi
```

To prevent a possible SQL injection from outside. This is easily demonstrated by:
``` bash
echo "QueryLimit=\"ABC\"" >> /etc/pihole/pihole-FTL.conf
pihole -q -listnew -all
```
and getting the error:
``` plain
  [i] Running query, please be patient...Error: near line 2: near "ABC": syntax error
  [i] No new domains have been found.
```

The injection could have been something worse, like disabling all adlists or injecting some (unwanted) whitelist domains.